addons/purchase/models/purchase.py

```diff
@@ -382,27 +317,6 @@ def button_confirm(self):
 
     @api.multi
     def button_cancel(self):
-        for order in self:
-            for pick in order.picking_ids:
-                if pick.state == 'done':
-                    raise UserError(_('Unable to cancel purchase order %s as some receptions have already been done.') % (order.name))
-            for inv in order.invoice_ids:
```

Comment:
@jem-odoo not sure that you should remove the check on invoice

Reply:
You are right; I'll keep it in purchase module.